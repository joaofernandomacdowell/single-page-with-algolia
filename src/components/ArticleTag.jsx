import React from 'react';

const ArticleTag = ({ tags, url }) => {
  const mainTag = tags[0];
  const taghref = url.substring(0, url.lastIndexOf('/'));

  return (
    <a className="zoom-article-tag" href={taghref}>
      {mainTag}
    </a>
  );
}

export default ArticleTag;