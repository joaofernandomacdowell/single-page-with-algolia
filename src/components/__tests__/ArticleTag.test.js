import React from 'react'
import { shallow } from 'enzyme';

import mockApiResponse from '../../helper';
import ArticleTag from '../ArticleTag';


describe('ArticleTag component', () => {
  const props = {
    tags: mockApiResponse.hits[0].tags,
    url: mockApiResponse.hits[0].url
  };
  const wrapper = shallow(<ArticleTag {...props} />);

  test('match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('have tag label with the main tag', () => {
    const label = wrapper.find('.zoom-article-tag');

    expect(label).toHaveLength(1);
    expect(label.text()).toBe(mockApiResponse.hits[0].tags[0]);
  });

  test('anchor tag with the correct href', () => {
    const label = wrapper.find('.zoom-article-tag');

    expect(label.props().href).toBe('https://www.zoom.com.br/jogos-xbox-one/deumzoom');
  });
});