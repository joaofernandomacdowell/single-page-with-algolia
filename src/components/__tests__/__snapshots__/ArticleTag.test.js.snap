// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleTag component match the snapshot 1`] = `ShallowWrapper {}`;
